hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -261,37 +258,33 @@ private boolean run(Operation op, long startNanos) {
 
         try {
             if (!metWithPreconditions(op)) {
-                return false;
+                return;
             }
 
-            if (op.isTenantAvailable()) {
-                op.pushThreadContext();
-                op.beforeRun();
-                call(op);
-            } else {
-                return true;
-            }
+            op.beforeRun();
+            call(op);
         } catch (Throwable e) {
             handleOperationError(op, e);
         } finally {
-            op.afterRunFinal();
-            if (publishCurrentTask) {
+             if (publishCurrentTask) {
```

Comment:
we are missing here the call to `op.afterRunFinal()`

Reply:
Fixed.